src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1259,21 +1261,28 @@ InferenceProfiler::Summarize(
     const std::map<cb::ModelIdentifier, cb::ModelStatistics>& start_status,
     const std::map<cb::ModelIdentifier, cb::ModelStatistics>& end_status,
     const cb::InferStat& start_stat, const cb::InferStat& end_stat,
-    PerfStatus& summary, uint64_t window_start_ns, uint64_t window_end_ns)
+    PerfStatus& summary, uint64_t window_start_ns, uint64_t window_end_ns,
+    bool clamp_window)
 {
   size_t valid_sequence_count = 0;
   size_t delayed_request_count = 0;
   size_t response_count = 0;
 
   // Get measurement from requests that fall within the time interval
   std::pair<uint64_t, uint64_t> valid_range{window_start_ns, window_end_ns};
-  uint64_t window_duration_ns = valid_range.second - valid_range.first;
   std::vector<uint64_t> latencies;
   std::vector<RequestRecord> valid_requests{};
   ValidLatencyMeasurement(
       valid_range, valid_sequence_count, delayed_request_count, &latencies,
       response_count, valid_requests);
 
+
+  if (clamp_window) {
+    std::tie(window_start_ns, window_end_ns) = ClampWindow(valid_requests);
```

Comment:
```suggestion
    auto [start, end] = ClampWindow(valid_requests);
```

You should be able to do this in C++ 17 to help with readability.